src/dataspec/base.py

```diff
@@ -596,11 +691,16 @@ def from_val(
         else:
             namedtuple_type = None  # type: ignore
 
+        def conform_tuple(v) -> Union[Tuple, NamedTuple]:
+            return ((namedtuple_type and namedtuple_type._make) or tuple)(
+                spec.conform(v) for spec, v in zip(specs, v)
+            )
+
         return cls(
             tag or "tuple",
             pred=pred,
             specs=specs,
-            conformer=conformer,
+            conformer=compose_conformers(conform_tuple, *filter(None, (conformer,))),
```

Comment:
This is repeated several times. Worth a function?

Reply:
It uses a function 😄